pgoapi/rpc_api.py

```diff
@@ -90,7 +93,7 @@ def _make_rpc(self, endpoint, request_proto_plain):
         request_proto_serialized = request_proto_plain.SerializeToString()
         try:
             http_response = self._session.post(endpoint, data=request_proto_serialized)
-        except requests.exceptions.ConnectionError as e:
+        except (requests.exceptions.ConnectionError, requesocks.exceptions.ConnectionError) as e:
```

Comment:
Mayby throw different exception for proxy error?

Then it could be catched in client and "new" proxy set..